### ISSUE TITLE
Update docs README with antora workflow.

### ISSUE THREAD

**Issue by dmuelle** [OpenLiberty/docs#2223]

**Comment by dmuelle** [comment]
Hi @lauracowen @Charlotte-Holt here is the first draft of the updated README for the antora workflow:

https://github.com/OpenLiberty/docs/tree/draft

It's based off the README from the blogs repo. I wasn't quite sure how to balance the instructions for internal writers vs. external contributors. For example, there are some internal docs (cheat sheet, guidance etc) that we consult but I don't know if they should be linked to from  an open source repo- especially since some are in the cloud connections wiki and soon migrating to unknown destinations.

I'll add another section to document the playbook/ publishing process once we take over that part of the process.

**Comment by dmuelle** [comment]
Actually since the docs playbook has its own repo, any info on that should go in the README there?

**Comment by lauracowen** [comment]
re playbook - would it make sense to just link from the docs repo instructions to the playbook instructions when it comes to the playbook steps?

**Comment by lauracowen** [comment]
re guidance - how much of that is really confidential or could it be posted externally on a wiki on the docs repo? Anyone submitting fixes/contributions externally would need to know some of that too?

For more general ID info that isn't external and isn't specific to OL docs, I think don't link that.

For the blogs repo, I think I'm going to write an internal Box note that gives an introduction to the different types of blog post in more detail. Then just the instructions on how to do it are posted externally in the readme.

**Comment by dmuelle** [comment]
I don't think any of it's necessarily confidential. I can move the guidances for example config and concept topics to the Git wiki and link to them there. There's a "Peer review cheat sheet" that is in the Connections wiki but it's more of a general thing and not specific to OL so I'll leave that out.

**Comment by lauracowen** [comment]
While I remember, a comment I've received about the blogs equivalent that might be relevant here:
- Authors section Step 4. Maybe add that they need to wait until the PR has been approved and merged before Step 5 (I've had two people point this out to me about the blogs draft build...).
- Similarly for any other PRs, though it may be obvious after this first time.

Other thoughts:
- "hwo the concerns" - typo
- should the reviewer also write a comment saying they've approved it? (as well as adding the label)
- if the doc is a task - think they should add the testing needed (or whatever it's called) label, then switch that for the other when it has been tested?
- I still dispute when they should create the staging PR (from a github perspective, there's no problem having PRs sitting waiting for approval). I think they should create it as soon as they're ready for it to be published (otherwise, it could get missed) but it shouldn't be merged until the day of publication. I see your point about which version it's to be published in - maybe that should be made explicitly in the issue/PR in some way - like "next version" or "not until 20.0.0.11" etc? But, that said, it depends how you want to run down the list of topics to be merged to staging - if you're happy with going off the issues board, that's fair enough (though I'm still concerned something could get missed without having PRs already created but I guess that's up to you).
- "To trigger a staging build, use the same procedure that you used in step 5 to trigger a draft build." - is it necessary to say that? You've given the travis link for staging - they'll recognise what they need to do when they get there.
- Step 12 - this doesn't quite make sense. A topic can be "ready to publish" (ie have done all the reviews on the backlog board) ages before it can be merged into `staging`. I think this is the part of the process I don't quite follow:
  - If you/Charlotte do the collecting together of everything that is to be published in this release, what will you work from? This reads like you're intending to work down the Ready to Publish column but what happens to issues that have been fully reviewed but the topics can't yet be moved to staging?

In general looks good though.

Are you going to add a section for the steps for the ID focals to follow next? Or are they going to all go in the playbook repo?

**Comment by dmuelle** [comment]
slack comments from @mbroz2 :

- get rid of the 'docs' header up top
- when referencing reviews, in addition to listing David & Charlotte, I'd recommend adding a link to https://github.com/orgs/OpenLiberty/teams/docs-maintainers/members as a fallback in case David & Charlotte can't be reached (or if the maintainers list changes over time)
- Note that the current state of docs only handles 'delivering docs content to unreleased/future versions of docs' and we still need to documentation around:
- Antora (it needs to be called and linked to and it might be worth mentioning at least the nav.adoc and to a lesser point antora.yml, and maybe about the overall structure (overview) of the repo/content)
- Making changes to released documents
- Overview of the branches (and their intent)
- How to do a release is another one (which would include updating the version in antora.yml), but that documentation should live in docs-playbook and @szvonek will do a first pass to create that documentation and hand it over to the docs team

**Comment by dmuelle** [comment]
Hi @lauracowen @mbroz2 
I've updated the draft to address your comments. I've also added new sections for `Updating an existing topic` and `Editing the docs navigation`, as well as an short overview of the branches and versioning.

## Laura's comments

- Authors section Step 4. Maybe add that they need to wait until the PR has been approved and merged before Step 5 **added approval info**
- "hwo the concerns" - typo **fixed**
- should the reviewer also write a comment saying they've approved it? **fixed**
- if the doc is a task - think they should add the testing needed (or whatever it's called) label, then switch that for the other when it has been tested? **fixed**
- I still dispute when they should create the staging PR- **updated to reflect the clarified procedure for `staging`**
- "To trigger a staging build, use the same procedure that you used in step 5 to trigger a draft build." - is it necessary to say that? You've given the travis link for staging - they'll recognise what they need to do when they get there.- **I left this as is for now. I don't think it's self evident that both `draft` and `staging` are triggered by the same build, especially since what you select is called `master`. I didn't know this until it was explained to me**
- Step 12 - this doesn't quite make sense. **updated procedure, hopefully more clear now**

## Michal's comments

- get rid of the 'docs' header up top **fixed**
- when referencing reviews, in addition to listing David & Charlotte, I'd recommend adding a link to https://github.com/orgs/OpenLiberty/teams/docs-maintainers/members as a fallback in case David & Charlotte can't be reached **updated**
- Note that the current state of docs only handles 'delivering docs content to unreleased/future versions of docs' and we still need to documentation around:
- Antora **added**
- Making changes to released documents **see new section**
- Overview of the branches (and their intent) **added to intro**
- How to do a release is another one **this should go in the docs playbook readme, when that is done I can link to it from here**

**Comment by lauracowen** [comment]
"To write and publish content in Open Liberty Docs, you work with only the following three branches" - isn't it four branches? Or three plus your own working branch per issue? I like the summary in general but it reads like the author writes the doc directly in `draft`.

Otherwise, I think it looks good. Be interesting to see how the nav branches approach goes.

**Comment by dmuelle** [comment]
@lauracowen good point- I revised the branches section so it starts the writers personal branch

https://github.com/OpenLiberty/docs/tree/draft

**Comment by lauracowen** [comment]
looks good

**Comment by dmuelle** [comment]
Is this ready to go to staging and vNext branches?

**Comment by lauracowen** [comment]
sure

**Comment by dmuelle** [comment]
The README is now updated on staging and vNext

https://github.com/OpenLiberty/docs
